### ISSUE TITLE
sync not working (set as available offline option not available)

### ISSUE THREAD

**Issue by cast-away** [nextcloud/android#3673]
### Actual behaviour
- selecting the options menu on a file is missing the previously available "set as available offline"
- now you have to manually sync the file to propagate local changes

### Expected behaviour
- "set as available offline" should be an available option
- then sync would happen automatically
 
### Steps to reproduce
1. navigate to a file
2. chose options from the menu

### Environment data
Android version: 8.0

Device model: Motorola + Samsung

Stock or customized system: stock

Nextcloud app version: 3.5.0

Nextcloud server version: 14.0.6

**Comment by tobiasKaminsky** [comment]
This changed with latest version, but was not communicated properly :-/
There is now a background job that syncs every downloaded file every 15min, when on wifi.

**Comment by cast-away** [comment]
Thanks Tobias, this is a neat improvement. IMHO the download + set as available offline was a little confusing.

The issue is it does not seem to work. I noticed one of my files, which was marked both as "set available offline" and as favourite was not being updated, that's when I actually realised the option was missing. I had to manually sync the file for it to work.

**Comment by derekblankmccoy** [comment]
> This changed with latest version, but was not communicated properly :-/
> There is now a background job that syncs every downloaded file every 15min, when on wifi.

I don't think the new 'sync' works. Downloaded a file, uploaded a new version via the web and the Android client changed the corner icon to a red 'x'. It never downloads the new version.

**Comment by mase76** [comment]
Similar here. Changing a file on the android device, it never gets synced to nc. I use it to sync my keepass database.

**Comment by jedwabny** [comment]
Even sync every 15 minutes is not very handy for local changes propagation if someone is waiting for our job.

What is the reason to wait 15 minutes for upload, if we are on wifi, instead of just upload immediately?

**Comment by derekblankmccoy** [comment]
I agree, mobile data is not an issue. Having files available as soon as possible is important. Maybe have an option to automatically sync files below 1MB?

**Comment by tobiasKaminsky** [comment]
There is a limitation on Android to scan/get notified by changing files. 
Therefore we decided to run a job every 15min to see if there is a changed file. This is a tradeoff between update speed vs. data usage vs. battery usage.

**Comment by jedwabny** [comment]
I've personally never observed any problems with auto-upload of "available offline" files. But of course it's only one person and three devices possitive impression. I could be not aware of problems. 
Anyway, maybe observing of just a few files (less number than the android limitation), recently opened by external app (from nextcloud open with menu), is an option instead of completely take the user right to decide?

Please understand, it's significant disconfort. At least if your file is outdated then you see red icon, but if you have local not-uploaded change done by external app then icon is still green and user can be convinced that his workflow is done.

Moreover, really sorry for saying this, taking into account the background jobs execution difficulties on various android devices and static time value in my humble opinion the new solution is step back.

**Comment by derekblankmccoy** [comment]
> There is a limitation on Android to scan/get notified by changing files.
> Therefore we decided to run a job every 15min to see if there is a changed file. This is a tradeoff between update speed vs. data usage vs. battery usage.

I see. In that case we need to look at why this 15 minute sync is not working.
I still believe a "Keep in sync" or "sync on/sync off" option for a folder is the right option.

**Comment by sciurius** [comment]
I have several Android clients all running 3.5.1.
On a Linux client, I have a file README.txt that automatically gets modified once per hour. It gets synchronized to the nextcloud server immedeately.
On all clients, the file README.txt has been downloaded manually.
After a number of hours it turns out that all clients except one automatically update the local copy of the file.

On the misbehaving client the following happens.
The initial (manual) download of the file README.txt goes well. After a while, when the file has been modified on the server and the client detects this, the file gets a red cross and there’s a notification “Download failed - Could not download README.txt”. When I use the nc client file browser and click on the file I get the same error. The only remedy is to remove the file locally and download it again, manually. But the next time the file is changed on the server the same download error occurs again on the client.
It looks as if the client has access rights to create files, remove files, read files, but is not allowed to overwrite existing files…

FWIW, this client is running on Asus Z300M with Android 7.0.

**Comment by sciurius** [comment]
No improvements with 3.6.0 beta.

**Comment by gsantner** [comment]
Synced txt files with nextcloud app, modified in other app multiple times these days. Nextcloud web app still says date from days ago. So this did not sync anything automatically at all for me, just downloaded  once manually, nothing uploaded automatically at all.
Also opened the NC app to ensure that it's not like the app is not running or could not create background stuff. Still old unchanged file in there (in online view).

Having Android 9,  NC app 3.5.1, battery optimization also turned off for that app.

**Comment by cast-away** [comment]
I am struggling to make sense of this...

1. a feature that detected files changed locally and immediately uploaded them to the server, albeit with a slightly confusing UI, is removed
2. a new feature, with a neater UI, is implemented in its place
3. the new feature is limited to detecting changes in files every 15 minutes only, and currently does not work

Can't we just get back to what worked in the first place please?

**Comment by TobiasDev** [comment]
Yeah, I would also like the auto sync (even on mobile data) for smaller files. Eg. I have a shared todoTXT file with things and now that no longer syncs until I'm on WiFi which is not ideal

**Comment by AllirionX** [comment]
I have the same use case as @TobiasDev : a todo.txt file shared between several devices that no longer syncs(unless I do it manually).
To have the option to push local changes to the server automatically and instantly would nice.

**Comment by cast-away** [comment]
same use case as @TobiasDev and @AllirionX, the small files suggestion could be a nice workaround, is this technically feasible?

**Comment by derekblankmccoy** [comment]
Not stale

**Comment by gsantner** [comment]
..

**Comment by cast-away** [comment]
Not stale

**Comment by gsantner** [comment]
.....

**Comment by TobiasDev** [comment]
Not stale

**Comment by derekblankmccoy** [comment]
Not stale

**Comment by gsantner** [comment]
yeah nobody cares, thanks for noting again

**Comment by jedwabny** [comment]
Not stale

**Comment by sciurius** [comment]
Not stale.

**Comment by tobiasKaminsky** [comment]
> To have the option to push local changes to the server automatically and instantly would nice.

This then would be something like a real 2way sync, which is planned, but needs a server side function to get it working.

**Comment by gsantner** [comment]
@tobiasKaminsky 
> This changed with latest version, but was not communicated properly :-/ There is now a
> background job that syncs every downloaded file every 15min, when on wifi.


I tried this again multiple times todays and this **does not work at all**.


* Env NextCloud Android 3.8.0 from F-Droid (non-beta), NC15, Android 9.
* Case 1: Sync file with with NextCloud (download). Edit file with other app. Wait ~30 minutes. Check nextcloud on browser, no change, still old file as it was previously there.
* Case 2: Same as 1, but also open NextCloud app from time to time, including a minute before checking out NC web. NC web: No change.
* Case 3: Same as 1, but click manually sync at that one changed file...and it syncs that one file.

Manually syncing a file is not really what I as user want todo and expect from a cloud sync app todo itself. Thats what the app currently cannot deliver and thats for me as a user a really big issue.

While the old solution, with marking as offline was not really a burner, it worked at least. I could mark my 10 favourite files, lets say todos, shoppinglists and they were in sync with my other clients automatically. This is currently impossible with NextCloud on Android.

> Google Play description
> * Keep your favorite files and folders synced

This issue is now open for about 8 months..I'm just wondering is there even interest in this very core feature of an sync app? I mean, if theres not enough time to get a new promising solution out that actually works...why not at least bring back something that works?

**Comment by sciurius** [comment]
Still doesn't work.

**Comment by sciurius** [comment]
Still doesn't work.

**Comment by sciurius** [comment]
Still doesn't work.

**Comment by asdrubal77** [comment]
Tested it all again today.

Uninstalled Nextcloud for android (3.9.2). Restarted the phone. Installed Nextcloud 3.10. In phone config options, removed nextcloud from battery optimization list and enabled nextcloud sync in account settings.

Then created a text file on my server, using firefox on my desktop. Synced the file in the mobile app. Then changed the file on the mobile app. Waited a few hours. Nothing happened. Then I manually synced (the only way to have files synced with nextcloud android app since version 3.5), and changed file on the server. Then waited again. And nothing happened again. 

Uninstalled version 3.10 and installed 3.4.2. Again, had to remove the app from the optimization list and activate sync on account settings.

Every time I changed the file on the mobile app, it was automatically uploaded to the server a few minutes later (even if nextcloud mobile app wasn't in the foreground). The same thing happened the other way: every time I changed the txt file on the server (using firefox on my desktop), the file on my phone was updated.

Have to mention that I didn't experiment problems with the auto-upload feature after 3.4.2 (not that I had any problem before that, as far as I remember). But I didn't do any testing.

**Comment by michaelhwalther** [comment]
not stale - at all...
Issues:
- opening a file requires "online" connection (doesn't work on flight mode)
- sync not reliable when online (as described)

**Comment by cast-away** [comment]
not stale

**Comment by derekblankmccoy** [comment]
Still not sure this 15 minute sync the devs talk about exists. If it does, it’s not working. The way I’m using Nextcloud requires that when I upload files to a shared folder, every user that I have shared that folder with gets those files downloaded onto their device as soon as possible and is available offline.

**Comment by jedwabny** [comment]
I still do not understand the 15 minutes interval idea. Desired behavior is to sync immediately - exactly like it already was implemented and now wiped out. Moreover I do not observe it working with the exception of one situation, where (today) one of Android devices creates duplicates exactly each 15 minut. Unfortunately it's pdf file and as a result I have a lot of following duplicates, and the number is growing:
File (2).pdf
File (2)(2).pdf
File (2)(2)(2).pdf
And so on.. 
Nextcloud server is 18.0.1 and Android client is the newest one from Play.

**Comment by sciurius** [comment]
not stale

**Comment by tobiasKaminsky** [comment]
> a. Desired behavior is to sync immediately - exactly like it already was implemented and now wiped out.

It was never immediately, but always a background job that synced it.
The only difference is that it is now a default behaviour for all downloaded files and it only happens on wifi.

**Comment by jedwabny** [comment]
Unfortunately it's not working so reliable as in the past worked for "offline mode" files.

I've just tried again. On Android, opened txt file by "open with" using QuickEdit, made some random modification, saved and exit from QuickEdit.

Now I'm waiting on Web GUI of NextCloud for more than half of hour and nothing is synced on server. On Android client the file has green icon attached so for user it looks to be ok - but it's not.

The new thing, that I observe, is found on Web GUI while trying to look into the file after changed on Android Client, the message appears: "! Document has been changed outside the editor. Changes can't be applied." (not the original but the translation to English).

Still manual sync is needed to push the changes done in external editor and also to fix the above message.

**Comment by asdrubal77** [comment]
> 
> 
> > a. Desired behavior is to sync immediately - exactly like it already was implemented and now wiped out.
> 
> It was never immediately, but always a background job that synced it.
> The only difference is that it is now a default behaviour for all downloaded files and it only happens on wifi.

But the question is not if the sync was immediately or not. I think that jedwabny didn't make himself very clear. Probably he meant "automatically"  instead of "immediately" .

The problem that all of us (who have commented on this ticket) have reported is that the sync feature (since version 3.5) doesn't sync! Even if you wait for hours! The only way to have files synced since version 3.5 is to manually sync them!

**Comment by michaelhwalther** [comment]
| ... the sync feature (since version 3.5) doesn't sync!

Exactly this!
It simply doesn't work anymore.

asdrubal77 <notifications@github.com> schrieb am Do., 23. Apr. 2020, 16:25:

**Comment by jedwabny** [comment]
Actually I meant both, "automatically" and also "immediately" (at least for pointed out files), because the declared 15 minutes interval between sync is quite inconvenient. Really, in the old versions, it synchronized to the server a second or two after quit the external editor.

But like the other said, it's not working at all for a year now.

**Comment by gsantner** [comment]
Comment I've written on 08. Nov 2019 is still the case, and so like all others say: Background sync doesnt work at all! neither "15 minutes" nor something else.

The previous "mark offline" stuff at least worked fine for single files, like above said some seconds afer modification.

I'm using Syncthing now since long time and recommend it to my app users, hope I can once recommend NC in future again. Syncthing just simply works, file modified, network connection available, boom file synced. 

But we all get it, there seems to be not much interest in getting automatic file sync to work again. Otherwise this wouldn't be stale over and over again.

**Comment by asdrubal77** [comment]
Well, there is hope. Tobias labeled this issue as "needs info/discussion".

**Comment by michaelhwalther** [comment]
> Well, there is hope. Tobias labeled this issue as "needs info/discussion".

... and then it went stale again...

Should there be a ticket saying: "Sync does not work" ?
The headline of this ticket only talks about a missing offline option... sounds a lot less intrusive than the actual problem...

**Comment by cast-away** [comment]
> Well, there is hope. Tobias labeled this issue as "needs info/discussion".
we have talking about this for a year and not too sure what additional information could be required but happy to provide it if pointed in the right direction
changed the title to grab more attention as suggested by@michaelhwalther

**Comment by cast-away** [comment]
oh dear, I clicked on the wrong button, how do I reopen it?

**Comment by asdrubal77** [comment]
@cast-away Maybe you could also update your first comment on Feb 2019, to make it clear that you also tested the then new sync feature and that it's not working.

**Comment by michaelhwalther** [comment]
ping - still an issue - still no resolution...

**Comment by strk** [comment]
@michaelhwalther did you test over wifi, and waiting 15 mins, after making the file available offline ?

While I do understand the 15minutes interval (still not working) I think making that time configurable would be nice, as well as the "only on wifi" part. I cannot test as I don't have wifi but seeing it NOT WORKING over data connection has been frustrating ...

**Comment by cast-away** [comment]
not stale as issue still unresolved

**Comment by tobiasKaminsky** [comment]
>  I cannot test as I don't have wifi but seeing it NOT WORKING over data connection has been frustrating ...

This is just a tradeoff between "always up to date" and "save data plan".
To be honest, if there would be an option to "always have everything up to date" users will enable it and then complain that it syncs too often, eats up data plan and drains battery…

**Comment by gsantner** [comment]
To be honest, sounds like a pretty weak point for not implementing sync to a cloud filemanager and file sync app. When looking at Syncthing and multiple shares.. nearly instant everything in-sync when editing on some device. And battery is still fine and can survive at least a whole day without charging phone.

**Comment by gsantner** [comment]
>  To be honest, if there would be an option to "always have everything up to date" users will enable it and then complain that it syncs too often, eats up data plan and drains battery…

To be honest, sounds like a pretty weak point for not implementing sync to a cloud filemanager and file sync app. When looking at Syncthing and multiple shares.. nearly instant everything in-sync when editing on some device. And battery is still fine and can survive at least a whole day without charging phone.  
It's good to have sane defaults that don't consume more battery than now and have the option for something else (where a attention notice is possible). And in any case it's always possible to improve the sync code/architecture to not drain battery. Other projects show that it's possible.

> To be honest, if there would be an option to "always have everything up to date" users will enable it

Oh no, people want to use the NextCloud files app to keep files in sync! Btw, on i.e. Google Play description says "file sync" and "keep XYZ in sync". Maybe remove that part?

--------

I mean, I get it many people reported sync doesn't work and it's ignored for long time. Theres not really a interest on it. For this there is other "the unix way" software which is specialized on full filesync. NextCloud isn't mainly targeting files anymore but trying to be everything...that needs to be accepted.

I'm now unsubscribing here as it's anyway inactivity ping-pong without changes only, which helps nobody.

**Comment by sciurius** [comment]
Sigh (also doesn't help)

**Comment by asdrubal77** [comment]
> > I cannot test as I don't have wifi but seeing it NOT WORKING over data connection has been frustrating ...
> 
> This is just a tradeoff between "always up to date" and "save data plan".
> To be honest, if there would be an option to "always have everything up to date" users will enable it and then complain that it syncs too often, eats up data plan and drains battery…

Even though there were complaints about the 15 minutes sync, the main complaint in this issue is that even the 15 minutes sync doesn't work! I myself am ok with the 15 minutes wait, although I understand that many need a faster sync.

So, just to make things clear: this is not about waiting 15 minutes to have your files in sync. It's about waiting forever, because the sync simply does not work at all!

**Comment by sciurius** [comment]
Sigh.

**Comment by mase76** [comment]
When marking a folder for syncing, I expect, that new files or folders inside the synced folder also get synced. For now only existing content is synced. New content is ignored.

**Comment by tobiasKaminsky** [comment]
> When marking a folder for syncing

This is a one-time sync only, so they are not permanently marked for sync.


> New content is ignored.

Yes, this is not yet implemented. We then also have to consider that such folder might grow so big that device can get out of storage…

**Comment by sciurius** [comment]
> We then also have to consider that such folder might grow so big that device can get out of storage

That is by far the weakest excuse I heard so far for not implementing this.

**Comment by tobiasKaminsky** [comment]
That should not be an excuse, it is more like a todo what we have to consider when implementing it.
The real "excuse" is that there is currently lack of time.

**Comment by cast-away** [comment]
still unresolved, far from stale

**Comment by cast-away** [comment]
not stale, still unresolved

**Comment by sciurius** [comment]
Not stale.

**Comment by Merrit** [comment]
This is still an issue.

**Comment by SeWieland** [comment]
Hmmm.. this is _still_ an issue :/

As it is labeled with "needs info/discussion", I will shortly describe why I need this very special feature of actually "syncing" files very badly (and I think a lot of users are in a very similar situation).

I use keepass / KeePassXC on my desktop devices. I cannot make use of the KeePassWeb app in nextcloud, because I use a second factor (who does not).
Furthermore, KeePassXC is way more convenient in usage and features (IMHO).
Plus, I'm not ever gonna change apps (to something incompatible to keepass), as I already have 300+ passwords and other secrets stored in this precious file.
However, syncing the database file actually works _really great_ using Nextcloud! :smiley: 
Thankfully those times are over where I had to merge X merge conflict databases manually :+1:

For mobile, I _could_ use keepass2andorid with WebDAV support, but I find KeepassDX a lot more useful and won't switch back.
The owner of this app however states (rightfully) that the app should _not_ be responsible for syncing the database, as it's out of scope.
This should be done by an app that has the _sole purpose_ of doing so.
I highly agree with that.

Quite some time back, the syncing using the android client _actually worked quite well_, until some update ruined that completely for me.
Since then, I'm passionately awaiting this feature to re-appear.
I cannot do _any_ changes on the android side, as syncing is _simply impossible_.

IMHO, from a user perspective, there _has to be_ an option to _immediately_ sync any files / folders upon _any_ changes.
Both - changes on the server and the client side (that's called synchronization).

I'm fully aware, that there might be some users that will misuse such an "expert feature" and try keeping the complete storage in sync with their phones.
In such a case, an "expert" mode that can be unlocked in the settings - with a proper warning - that unlocks the "keep in sync" feature, might be of help.

**In my case, I only would use it for exactly _1 file_. (And yes, I want to sync that few KiloBytes immediately without WiFi as well.)**

Just for the record: My personal nextcloud is alive since back in the "owncloud 7" days. And because of not being able to sync that single file, I'm really considering "syncthing" right now...

**Comment by gsantner** [comment]
forget it & dont waste your time, it has been described numerous times here. they dont really care otherwise it would have been looked into properly for ages.

I switched to syncthing and never looked back. I would rather say NC is a  cloud file manager which happens to have manual download/upload capabilities, rather a "file & data sync solution".

**Comment by Merrit** [comment]
:disappointed: This is still an issue.

**Comment by sciurius** [comment]
Fighting stale bots doesn't really keep me motivated to participate.

**Comment by aurisnoctis** [comment]
Syncing doesn't work at all using the Nextcloud app on Android.

I waited several weeks and my simple little todo.txt on my Android with Nextcloud app is never uploaded to my Nextcloud instance hosted at webo.hosting. I have to remember to manually trigger an upload from within my Android Nextcloud client.

I am on WiFi all the time. Syncing doesn't work. Not even after weeks.

**Comment by aurisnoctis** [comment]
I noticed that the original first issue post hast very few thumbs up. Maybe anyone interested in drawing attention to this issue use the 👍 feature on the top post.

**Comment by thstyl2000** [comment]
Please consider setting this into higher priority! Primary use for me is to sync a number of pdf files (student submissions), correct them on my tablet and have my corrections synced and backed up to my work folder. Does the notify_push app come in here handy?

**Comment by Merrit** [comment]
😔

**Comment by aurisnoctis** [comment]
Update for my https://github.com/nextcloud/android/issues/3673#issuecomment-808814774: automatic one-way update works for me now.

**Setup** (with my own translation from German UI):
* Nextcloud App > Settings > automatic Upload
*  3-dot menu in top right > create own folder > choose local folder (e.g., where todo.txt is) + deactivate 'only via WiFi' + 'overwrite on Server if it exists already' > save

It's still not a real sync, of course - only works if changes are made on phone only, never on server, but works in background and uploads automatically.

**Comment by cast-away** [comment]
this has not been addressed

**Comment by cast-away** [comment]
not resolved @aurisnoctis' workaround could be a good enough workaround for some but as he explains this is not full synchronization as it only works one way: from phone to server.

**Comment by Zacho2** [comment]
not stale

**Comment by cast-away** [comment]
not stale

**Comment by aurisnoctis** [comment]
probably still not stale

**Comment by AlvaroBrey** [comment]
> probably still not stale

I've disabled the stale-bot for this issue, we'll take a look at it whenever we can as we're spread thin. Sorry!
